MaxText/checkpointing.py

```diff
@@ -64,6 +72,53 @@ def create_orbax_checkpoint_manager(
   return mngr
 
 
+def create_orbax_emergency_checkpoint_manager(
+    local_checkpoint_dir: str,
+    persistent_checkpoint_dir: str,
+    global_mesh: jax.sharding.Mesh,
+    abstract_state: PyTree,
+    local_save_interval_steps: int,
+    persistent_save_interval_steps: int,
+):
+  """Returns an emergency checkpoint."""
+  max_logging.log("Creating emergency checkpoint manager...")
+
+  local_registry = type_handlers.create_type_handler_registry(
+      (
+          jax.Array,
+          type_handlers.ArrayHandler(primary_host=None, replica_id=None),
+      ),
+  )
+
+  local_checkpoint_handler = PyTreeCheckpointHandler(
+      use_ocdbt=True,
+      use_zarr3=True,
+      primary_host=None,
+      type_handler_registry=local_registry,
+  )
+
+  options = emergency_checkpoint_manager.CheckpointManagerOptions(
+      local=LocalCheckpointOptions(
+          save_interval_steps=local_save_interval_steps
+      ),
+      persistent=PersistentCheckpointOptions(
+          save_interval_steps=persistent_save_interval_steps
```

Comment:
should we have 
`enable_async_checkpointing=use_async`
and
`create=True` just like the regular checkpoint manager?

https://github.com/google/maxtext/blob/9df576142bba46ea539c6ff6fe63e743d7b7c12c/MaxText/checkpointing.py#L55C29-L56C1

Reply:
Regarding `enable_async_checkpointing=use_async`, for any workload large enough to need this feature, they wouldn't want to use sync checkpointing, so we are making the decision for them and simplifying their experience.

On `create=True`, great question! Due to some Orbax limitation, the persistent directory has to be created by the user first.